icu4c/source/data/buildtool/utils.py

```diff
@@ -86,6 +86,14 @@ def get_all_output_files(requests, include_tmp=False):
     return [f for _, f in set((type(f), f) for f in files)]
 
 
+def compute_directories(requests):
+    dirs = set()
+    for file in get_all_output_files(requests, include_tmp=True):
+        path = "%s/%s" % (dir_for(file), file.filename)
+        dirs.add(path[:path.rfind("/")])
```

Comment:
This together with the previous line is not a no-op because the file.filename may contain `/` right?
Why not use a directory function from the os package?

Reply:
Correct.

I handle all filenames everywhere using / and string operations.  I don't touch os except at the boundary.  If you don't like this, a way to make the code a little bit cleaner would be to convert the InFile, TmpFile, and OutFile namedtuples to classes and add some sugar methods.  Or, I wonder if the sugar methods can be attached to the namedtuples directly.  Then at least we put all of the string-based operations in one place.